src/software/thunderscope/chicker/chicker.py

```diff
@@ -0,0 +1,227 @@
+import pyqtgraph as pg
+import software.thunderscope.constants as constants
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import (
+    QApplication,
+    QCheckBox,
+    QGridLayout,
+    QGroupBox,
+    QButtonGroup,
+    QMenu,
+    QPushButton,
+    QRadioButton,
+    QVBoxLayout,
+    QHBoxLayout,
+    QWidget,
+    QSlider,
+)
+
+
+class ChickerWidget(QWidget):
+    def __init__(self, parent=None):
+        super(ChickerWidget, self).__init__(parent)
+
+        grid = QGridLayout()
+        self.radio_buttons = QButtonGroup()
+        # grid --> groupBox --> button/slider
+        grid.addWidget(self.createButton("Charge"), 0, 0)
+        grid.addWidget(self.createButton("Kick"), 0, 1)
+        grid.addWidget(self.createButton("Chip"), 0, 2)
+        grid.addWidget(self.createRadio("No Auto"), 1, 0)
+        grid.addWidget(self.createRadio("Auto Kick"), 1, 1)
+        grid.addWidget(self.createRadio("Auto Chip"), 1, 2)
+        grid.addWidget(self.createSlider("Geneva Slider", 1, 5, 1), 2, 0, 1, 3)
+        grid.addWidget(self.createSlider("Power Slider", 1, 100, 10), 3, 0, 1, 3)
+        self.setLayout(grid)
+        self.grid = grid
+
+        # state of radio buttons in order NoAuto, AutoKick, AutoChip
+        self.radioCheckable = [True, True, True]
+        self.charged = False
+        self.geneva_value = 1
+        self.power_value = 1
+
+    # all the buttons, radios, and sliders are in a group box --> vbox (vertical alignment)
+    def createButton(self, text):
+        groupBox = QGroupBox()
+        button = QPushButton(text)
+        groupBox.setStyleSheet("color: black")
+        button.setCheckable(True)
+        if text != "Charge":
+            button.setCheckable(False)
+            button.setStyleSheet("background-color: Grey")
```

Comment:
lets pass in a "disabled_by_default" argument to this function and use this here so its generic to all widgets